_projects/go_bears_on_bsd.md

```diff
@@ -0,0 +1,102 @@
+---
+collaborating_projects:
+  - coala
+desc: "Explore improvements to coala's dependency management through having Go Linters installed in *BSD systems automatically."
+difficulty: medium
+initiatives:
+  - GSoC
+issues:
+  - "https://gitlab.com/coala/package_manager/issues/149"
+  - "https://github.com/coala/coala-bears/issues/2481"
+markdown: go_bears_on_bsd.md
+mentors:
+  - xferra
+  - yukiisbored
+name: "Linter Dependency Management from a Bare-Metal System"
+requirements:
+  - "At least one patch to the coala core should be accepted and merged."
+  - "The proposal needs to show familiarity with coala's dependency management implementation."
+status: in_progress
+tags:
+  - Automation
+  - Installation
+  - Dependency
+---
+
+`coala-bears` are of two types, Native Bears which implement all the business
+logic themselves and Linter Bears which wrap existing linters and checkers.
+Dependencies of bears are mostly language runtimes, language-specific packages
+(e.g. python:nltk) or command line tools (e.g. git). Both Native Bears and
+Linter Bears have dependencies which need to be installed before they can be
+executed.
+
+`dependency_management` aims to provide Python APIs for dependency management on
+various languages and different systems. It is used extensively by coala,
+however, it lacks most of the key implementation and documentation in its
+current state.
+
+coala now has [several bears written in
+Go](https://github.com/coala/coala-bears/search?q=GoRequirement), [7 of
+which](https://github.com/coala/coala-bears/blob/master/bears/go/) are for
+validating Go files. They need specific versions of Go and some externally
+supplied packages, and currently Go linter bears just assume they are all in
+place and work. Go language dependencies are currently managed unversioned in
+coala-bears. There is no official way provided for versioning, and though there
+are other commonly used mechanisms for installing specific versions of Go repos,
+a lot more investigation is needed.
+
+The end goal is to be able to bootstrap and recursively install all dependencies
+on any systems without anything installed other than a base system, but the
+scope of a complete solution is too large. A proposal has been made with
+suitable scope, which is to automate the installation of Go linter bears onto
+bare-metal BSDs.
+
+Features will mainly be implemented to coala's `dependency_management`. It
+should be able to operate system and language-specific package managers to
+handle dependencies. It should also resolve package names and versions
+automatically using Repology on multiple operating systems.
```

Comment:
Do you plan to use Repology as a service or cache it's data somehow? It's it possible that some package names will not be resolved?

Reply:
I believe this partially lies in @nemaniarjun 's project scope which is to implement Repology as a service. What I'd like to do is to make use of the infrastructure provided by that project and specifically implement and ensure the dependencies required for my project has correct data from Repology.